cmd/go-clang-generate/generate/enum.go

```diff
@@ -156,77 +143,80 @@ func (e *Enum) addEnumStringMethods() error {
 	return nil
 }
 
-var templateGenerateEnumSpelling = template.Must(template.New("go-clang-generate-enum-spelling").Parse(`
-func ({{$.Receiver}} {{$.ReceiverType}}) Spelling() string {
-	switch {{$.Receiver}} {
-		{{range $en := $.Cases}}case {{range $sn := $en.CaseStr}}{{$sn}}{{end}}: return "{{$en.PrettyStr}}"
-		{{end}}
-	}
+func (e *Enum) addEnumSpellingMethod() error {
+	f := newFunction("Spelling", e.Name, "", "", Type{GoName: "string"})
+	fa := newASTFunc(f)
 
-	return fmt.Sprintf("{{$.ReceiverType}} unkown %d", int({{$.Receiver}}))
-}
-`))
+	fa.generateReceiver()
 
-func (e *Enum) addEnumSpellingMethod() error {
-	var b bytes.Buffer
+	fa.addReturnType("", Type{
+		GoName: "string",
+	})
 
-	type Case struct {
-		CaseStr   []string
-		PrettyStr string
-	}
+	switchStmt := doSwitchStmt(&ast.Ident{Name: f.Receiver.Name})
+	fa.Body.List = append(fa.Body.List, switchStmt)
 
-	type Switch struct {
-		Receiver     string
-		ReceiverType string
-		Cases        []Case
-	}
+	m := make(map[uint64]*ast.CaseClause)
 
-	s := &Switch{
-		Receiver:     e.Receiver.Name,
-		ReceiverType: e.Receiver.Type.GoName,
-		Cases:        []Case{},
-	}
+	for _, enumerator := range e.Items {
+		if caseClause, ok := m[enumerator.Value]; !ok {
```

Comment:
Yes, exactly.
Further, Go doesn't like multiple definitions, even in the same clause.